### DISCUSSION TITLE
Using multi-view feature from ImGui docking branch on Windows

### DISCUSSION THREAD

**Discussion by yangcshen** [vsg-dev/VulkanSceneGraph#798] [General]
This is not a feature request per se but just a question. I would like to use multi-view feature from ImGui docking branch. I forked vsgImGui and used the docking branch from ImGui, then rebuilt it and the docking works fine. But I want to drag imgui widgets outside of the window created by vsg, which is already integrated by ImGui if multiview flag is enabled. The issue is that this feature is not supported on win32 platform out of the box.  And vsg is using win32 API to create windows on Windows platform.

See this discussion (https://github.com/ocornut/imgui/discussions/5871).  I do not even know where ImGui::GetPlatformIO().Platform_CreateVkSurface is invoked. I put breakpoint in my code but this function is never called. 

Can anyone give me some instructions on how to implement this handle or point me to the right direction?
The end goal is to have ImGui UI widgets which can be dragged and moved outside of the window. 

Thanks,
Cameron

**Comment by robertosfield** [comment]
I have moved this from VulkanSceneGraph Issue to Discussion as it's not a bug with the VulkanSceneGraph, but an question about how one should implement novel ImGui features with vsgImGui.

I have only dabbled with ImGui so far, and didn't even know that it might be possible to having it support dragging widgets off the main window, so I don't have an specifics insights to add.

As a general note, if the dock feature is experimental part of ImGui then it may be worth waiting till it's part of a main release.  As for vkCreateSurface, this is something that happens after the actual Window has been created and is done by the various vsg::Window subclasses.  If you were to create a new window on demand then one would need to go through the static vsg::Window::create(..) method.  

If ImGui is creating the Window but then wants the application code to create the vkSurface then one might need to get ones hands dirty with getting the required Window handles from ImGui.

That's as much as I can provide as input.  Most of the legwork will need to be done yourself unless others wish to see this feature supported and can chip in time themselves.
